xs2a-adapter-service-remote/src/test/java/de/adorsys/xs2a/adapter/remote/service/impl/RemoteOauth2ServiceTest.java

```diff
@@ -0,0 +1,83 @@
+package de.adorsys.xs2a.adapter.remote.service.impl;
+
+import de.adorsys.xs2a.adapter.api.model.HrefType;
+import de.adorsys.xs2a.adapter.remote.api.Oauth2Client;
+import de.adorsys.xs2a.adapter.rest.api.model.TokenResponseTO;
+import de.adorsys.xs2a.adapter.service.Oauth2Service;
+import de.adorsys.xs2a.adapter.service.model.TokenResponse;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.doReturn;
+
+class RemoteOauth2ServiceTest {
+
+    private Oauth2Client oauth2Client = Mockito.mock(Oauth2Client.class);
+    private final RemoteOauth2Service service = new RemoteOauth2Service(oauth2Client);
+
+
+    @Test
+    void getAuthorizationRequestUri() throws IOException {
+        Oauth2Service.Parameters parameters = new Oauth2Service.Parameters();
+        String authUrl = "http://localhost:8080/auth";
+
+        doReturn(buildHref(authUrl)).when(oauth2Client).getAuthorizationUrl(anyMap(), any());
+
+        URI uri = service.getAuthorizationRequestUri(new HashMap<>(), parameters);
+
+        assertThat(uri).isEqualTo(URI.create(authUrl));
+    }
+
+    @Test
+    void getAuthorizationRequestUriWithException() throws IOException {
+
+        doReturn(buildHref("blablabla_unexisting_uri@2#^"))
+            .when(oauth2Client)
+            .getAuthorizationUrl(anyMap(), any());
+
+        assertThrows(IOException.class, () -> service.getAuthorizationRequestUri(new HashMap<>(),
```

Comment:
violates https://rules.sonarsource.com/java/RSPEC-5778

Reply:
adjusted, thanks

Reply:
please remove curly braces in the lambda, it complains about that too

Reply:
removed